opendrift/models/basemodel.py

```diff
@@ -1753,11 +1753,11 @@ def seed_within_polygon(self, lons, lats, number=None, **kwargs):
         area = 0.0
         for i in range(-1, len(x)-1):
             area += x[i] * (y[i+1] - y[i-1])
-        area = abs(area) / 2
 
         # Make points, evenly distributed
         deltax = np.sqrt(area/number)
         lonpoints = np.array([])
+        area = abs(area) / 2
```

Comment:
Seems generally good, but 4 tests are failing, presumably because this line has been moved, and thus affecting the area and giving the wrong number of elements. Is this a mistake?